### PR TITLE
vdk-jupyter: ui end-to-end testing

### PR DESCRIPTION
**Pull request by duyguHsnHsn**
What:
Introduced tests for currently existing ui components - vdk menu and its buttons.

Why:
Only manual tests were done so far.

Signed-off-by: Duygu Hasan [hduygu@vmware.com](mailto:hduygu@vmware.com)

### CODE REVIEW

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts [hunk 1]
```diff
@@ -1,21 +1,125 @@
 import { expect, test } from '@jupyterlab/galata';
+import { assert } from 'console';
 
 /**
  * Don't load JupyterLab webpage before running the tests.
  * This is required to ensure we capture all log messages.
  */
 test.use({ autoGoto: false });
+const delay = ms => new Promise(res => setTimeout(res, ms));
```
Comment (ivakoleva) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts:9]: nit: so the tests are grouped logically by some criteria,
for example user input validation tests.
Then we have happy path coverage, like download job then verify job had been downloaded the way expected

**Review by ivakoleva — APPROVED** [review]
LGTM

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/commandsAndMenu.tsx [hunk 2]
```diff
@@ -62,7 +62,9 @@ export function updateVDKMenu(commands: CommandRegistry) {
           buttons: [Dialog.okButton(), Dialog.cancelButton()]
         });
         if (!result.value) {
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/commandsAndMenu.tsx:64]: Nit: Consider doing something like

```
const RESULT_BUTTON_CLICKED = !result.value && result.button.accept;
if (RESULT_BUTTON_CLICKED) {
  createJobRequest();
}

```

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts [hunk 3]
```diff
@@ -1,21 +1,125 @@
...
 
-test('should emit an activation console message', async ({ page }) => {
-  const logs: string[] = [];
+test('should open run job pop up and then cancel the operation', async ({ page }) => {
+  await page.goto("");
+  await page.menu.open('VDK');
+  await page.locator('#jp-vdk-menu').getByText('Run').click();
+  await page.locator('div').filter({ hasText: 'Run Job' });
+  await page.getByRole('button', { name: 'Cancel' }).click();
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts:16]: yes, if the last one is executed without errors this means that the pop up is closed

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts [hunk 4]
```diff
@@ -1,21 +1,125 @@
...
+  await page.locator('div').filter({ hasText: 'Run Job' });
+  await page.getByRole('button', { name: 'OK' }).click();
+  await page.locator('div').filter({ hasText: 'Encountered an error when trying to run the job.' });
+  await page.getByRole('button', { name: 'OK' }).click();
+
+});
 
-  page.on('console', message => {
-    logs.push(message.text());
+test('should try to run a job with incorrect data and get a dialog error message', async ({page}) =>{
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts:31]: This is completely unrelated, but shouldn't the bot be configured to re-format these? There are other places in the code where with no space before the opening brace of a function and the bot just lets it slide. 🤔

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/commandsAndMenu.tsx (hunk 2 above)
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/commandsAndMenu.tsx:64]: good point, will change it

**Review by DeltaMichael — APPROVED** [review]

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts (hunk 4 above)
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts]: I fixed it manually to push merge this PR, but the topic should be discussed further.

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts (hunk 1 above)
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/ui-tests/tests/vdk-jupyterlab-extension.spec.ts]: Further tests should be added. I will push these ones for now,  and the others can be discussed latter while adding other tests for the functionalities. Current tests mainly focus on the JSX components that are being created.

**Pull request merged**